### PR TITLE
Closes AgileVentures/MetPlus_tracker#428

### PR DESCRIPTION
**Pull request by patmbolger**
This story includes these tasks:

- [x] Add mailgun public API key to application.yml
- [x] Create initializers/email_validate_service.rb to define the service base URL , including API version to use
- [x] Create app/services/email_validate_service.rb to invoke mailgun service and return validation results
- [x] Edit validators/app/validators/email_validator.rb to call service method and return validation error as appropriate
- [x] Add email validation to user.rb model (augment Devise validation)
- [x] Add service stub method(s) to spec/support/service_spec_helpers.rb
- [x] Create spec/services/email_validate_service_spec.rb
- [x] Stub out other Spec tests as needed
- [x] Convert email addresses used in acceptance tests to valid addresses, as needed

Adding cucumber steps/tests, to confirm email validation errors, will be done via a separate story.

### CODE REVIEW

File: config/initializers/email_validate_service.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+EMAIL_VALIDATE_URL = 'api.mailgun.net/v3'
+
+MAILGUN_PUBLIC_KEY = 'pubkey-0188160a7f79dbc9c673c5c251915dcf'
```
Comment (joaopapereira) [inline config/initializers/email_validate_service.rb]: Should we store this here or in a application.yml variable?

File: spec/models/agency_spec.rb [hunk 2]
```diff
@@ -47,13 +47,16 @@
...
      end
 
      describe 'Email' do
        subject {FactoryGirl.build(:agency)}
-       it { should_not allow_value('asd', 'john@company').for(:email)}
+       it do
+         stub_email_validate_error
+         should_not allow_value('asd', 'john@company').for(:email)
+       end
        it { should allow_value('johndoe@company.com').for(:email)}
```
Comment (joaopapereira) [inline spec/models/agency_spec.rb]: Shouldn't we stub error or success in this one?

File: spec/models/company_spec.rb [hunk 3]
```diff
@@ -63,8 +63,14 @@
...
-       it { should_not allow_value('asd', 'john@company').for(:email)}
+       let!(:company) {FactoryGirl.build(:company)}
+       it do
+         stub_email_validate_error
+         company.email = 'asd'
+         expect(company).not_to be_valid
+         company.email = 'john@company'
+         expect(company).not_to be_valid
+       end
        it { should allow_value('johndoe@company.com').for(:email)}
```
Comment (joaopapereira) [inline spec/models/company_spec.rb]: Shouldn't we stub error or success in this one?

File: spec/models/company_spec.rb [hunk 4]
```diff
@@ -89,7 +95,10 @@
 
      describe 'Job Email' do
        it { should validate_presence_of(:job_email) }
-       it { should_not allow_value('asd', 'john@company').for(:job_email)}
+       it do
+         stub_email_validate_error
+         should_not allow_value('asd', 'john@company').for(:job_email)
+       end
        it { should allow_value('johndoe@company.com').for(:job_email)}
```
Comment (joaopapereira) [inline spec/models/company_spec.rb]: Shouldn't we stub error or success in this one?

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -54,30 +55,51 @@
...
+               '1 231 231 2345', '12312312345',  '1231231234',
                '1-910-123-9158 x2851', '1-872-928-5886', '833-638-6551 x16825').for(:phone)}
-     end  
+     end
+
+     describe 'Email validation' do
+       let(:user) { FactoryGirl.create(:user) }
+
+       it 'valid email address' do
+         user.email = 'thisone@yahoo.com'
```
Comment (joaopapereira) [inline spec/models/user_spec.rb]: Shouldn't we stub error or success in this one?

File: spec/services/email_validate_service_spec.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe EmailValidateService, type: :model do
+
+  describe 'email address is not specified' do
+    it 'returns invalid address' do
+      result = {status: 'SUCCESS', valid: false, did_you_mean: nil}
+      expect(EmailValidateService.validate_email('')).to eq result
```
Comment (joaopapereira) [inline spec/services/email_validate_service_spec.rb]: We should check API not called

File: spec/services/email_validate_service_spec.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    end
+  end
+
+  describe 'email address is valid' do
+    it 'returns valid address' do
+      result = {status: 'SUCCESS', valid: true, did_you_mean: nil}
+
+      expect(EmailValidateService.validate_email('goodone@gmail.com')).to eq result
```
Comment (joaopapereira) [inline spec/services/email_validate_service_spec.rb]: Is this being stubbed?

File: spec/services/email_validate_service_spec.rb (hunk 6 above)
Comment (patmbolger) [inline spec/services/email_validate_service_spec.rb]: Added an expectation for that.

File: spec/models/agency_spec.rb (hunk 2 above)
Comment (joaopapereira) [inline spec/models/agency_spec.rb]: Ok cool

**Comment by joaopapereira** [comment]
Going to merge

**Comment by joaopapereira** [comment]
Cannot merge on the phone so when I very to a pcampos@cm-aveiro.pt I will do it

**Pull request merged**